rio/src/RIO/Process.hs

```diff
@@ -258,6 +266,10 @@ exeSearchPathL = processContextL.to pcPath
 
 -- | Create a new 'ProcessContext' from the given environment variable map.
 --
+-- On Windows only, if the system @PATHEXT@ environment variable is absent from
```

Comment:
This isn't really true. This implies that a new `PATHEXT` value is added to the env vars. My original concern still also stands: I don't believe this function is the right place for this explanation, it should go exclusively on an identifier which is related to the path extensions.

Reply:
Which other exported identifier are you thinking of? The constructors of type `ProcessContext` are not exported and this behaviour is introduced by the `mkProcessContext` 'smart' constructor function. There is no function that manipulates the `pcPath` field other than via `mkProcessContext`.

(As an aside, I don't understand why `mkProcessContext` behaves in this way. A Windows user could legitimately - if unusually -  want to remove `PATHEXT` from the environment so as not to have the functionality on Windows that `PATHEXT` facilitates.)

In terms of the documentation itself, instead of:

"On Windows only, if the system `PATHEXT` environment variable is absent from the given map, `mkProcessContext` behaves as if the variable were present with its default value.`

how about:

"On Windows only, if the system `PATHEXT` environment variable is absent from the given map, `mkProcessContext` behaves as if the variable were present with its default value, but without adding it to the environment variables of the process context.`

Reply:
I wrote up a response to this which seems to have gotten lost, sorry.

I think this comment is approaching this backwards. It's saying we pretend like there's a field present, except for this one thing. Except the one thing is the only functionality (setting env vars) that people think it should be relevant to. It leaves open a bunch of questions to the reader, _especially_ the reader on a non-Windows machine, because they don't have the context of this problem in their mind.

Instead: I'd rather see the functions that actually look up executables say directly: on Windows, we use PATHEXT, and if it isn't there, we fall back to a default. I'd rather have that documentation in one place, ideally on the new `exeExtensions` accessor. And then I'd like to see it referenced from the relevant places. Right now, the concepts are spread across too many pieces of documentation, especially where it isn't relevant to most people.

Reply:
Thank you. I hope my revised pull request now reflects that.